graders/python/python_autograder/code_feedback.py

```diff
@@ -334,6 +334,81 @@ def bad(msg: str) -> Literal[False]:
 
         return True
 
+    @classmethod
+    def check_dict(
+        cls,
+        name: str,
+        ref: dict[Any, Any],
+        data: dict[Any, Any],
+        *,
+        only_keys: None | list[str] = None,
+        accuracy_critical: bool = False,
+        report_failure: bool = True,
+    ) -> bool:
+        """
+        Checks that a student dict (`data`) has all correct key-value mappings with respect to a reference dict (`ref`).
+
+        Parameters:
+            name: Name of the dict that is being checked. This will be used to give feedback.
+            ref: Reference dict.
+            data: Student dict to be checked. Do not mix this up with the previous dict! This argument is subject to more strict type checking.
+            only_keys: If not None, it will only only compare the keys listed.
+            accuracy_critical: If true, grading will halt on failure.
+            report_failure: If true, feedback will be given on failure.
+        """
+
+        def bad(msg: str) -> Literal[False]:
+            if report_failure:
+                cls.add_feedback(msg)
+            if accuracy_critical:
+                cls.finish("")
+            else:
+                return False
+
+        if data is None:
+            return bad(f"{name} is not a dict, got None")
```

Comment:
This seems a little redundant with the case below? I recognize the error message is a little nicer, but I don't really see this  specific case being a common error.

Reply:
Fixed in b0154849a9187cec010fdefb137a06043532db88.